### PR TITLE
[DB Manager] Fix delete raster layer in GeoPackage (Fix #57751)

### LINKED ISSUE #57751: Deleting raster from geopackage does nothing
### What is the bug or the crash?

I have that gpkg with a raster inside. 
[dummy aree allagabili.zip](https://github.com/user-attachments/files/15806054/dummy.aree.allagabili.zip)
I'm not able to delete it / rename from DBManager.
immagine
immagine

No error, but nothing happens.

If you open the file with DB Browser, all the data is still inside it.
immagine

Thanks!

### Steps to reproduce the issue

1. connect the file in the DB Manager
2. try to delete / rename the raster

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.34.7-Prizren | QGIS code revision | 6f7d735c
-- | -- | -- | --
Qt version | 5.15.13
Python version | 3.12.3
GDAL/OGR version | 3.9.0
PROJ version | 9.4.0
EPSG Registry database version | v11.004 (2024-02-24)
GEOS version | 3.12.1-CAPI-1.18.1
SQLite version | 3.45.1
PDAL version | 2.6.3
PostgreSQL client version | 16.2
SpatiaLite version | 5.1.0
QWT version | 6.2.0
QScintilla2 version | 2.14.1
OS version | Windows 11 Version 2009
  |   |   |  
Active Python plugins
AutoLayoutTool | 1.1
changeDataSource | 3.1
data_exporter | 0.1
firstaid | 3.1.1
FreehandRasterGeoreferencer | 0.8.3
GroupStats | 2.2.7
Group_and_Sort_Layers | 1.2
group_transparency | 0.2
layout_loader | 1.1
layout_panel-main | 0.3
MultipleLayerSelection | 2.0.1
OSMDownloader | 1.0.3
profiletool | 4.2.6
ProjectPackager | 0.7.2
QGISSortAndNumber-master | 1.0
QPackage | 1.5
QuickOSM | 2.2.3
quick_map_services | 0.19.34
quick_shape_update | 0.1
RimoPrinting_QGIS3 | 1.0
searchlayers | 3.0.16
db_manager | 0.1.20
grassprovider | 2.12.99
processing | 2.12.99

</body></html>

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by agiudiceandrea**
## Description

Fixes the "Delete..." functionality for raster layers in GeoPackage containers.

Fixes #57751.

Looking at the reported issue, I noticed that also the "Rename..." functionality doesn't work for raster layers in GeoPackage containers (it is not possible to rename a raster layer in a GeoPackage container even using the Browser panel https://github.com/qgis/QGIS/issues/30907).
@elpaso it looks like a `renameRasterTable` function is missing in `QgsGeoPackageProviderConnection`.
@rouault, is there a "simple" way to rename a raster layer in a GeoPackage? (it seems to me `ALTER TABLE ... RENAME TO ...` doesn't correctly work for raster layers in GeoPackage)

~~UPDATE: I've also removed the Rename action for raster layer in a GeoPackage.~~
UPDATE: I'll submit a new PR to fix the Rename action for raster layer in a GeoPackage.

### CODE REVIEW

**Comment by elpaso** [comment]
See: https://github.com/qgis/QGIS/issues/30907#issuecomment-2164970775

**Comment by agiudiceandrea** [comment]
> See: https://github.com/qgis/QGIS/issues/30907#issuecomment-2164970775
> > This is not yet supported by GDAL.

I've also removed the Rename action for raster layer in a GeoPackage.

**Comment by rouault** [comment]
> @rouault, is there a "simple" way to rename a raster layer in a GeoPackage?

no, it isn't supported currently. I've filed https://github.com/OSGeo/gdal/issues/10201 about that

File: python/plugins/db_manager/db_model.py [hunk 1]
```diff
@@ -398,7 +398,9 @@ def flags(self, index):
         if index.column() == 0:
             item = index.internalPointer()
 
-            if isinstance(item, SchemaItem) or isinstance(item, TableItem):
+            if isinstance(item, SchemaItem) \
+                    or (isinstance(item, TableItem)
+                        and not (self.hasGPKGSupport and item.getItemData().type == Table.RasterType)):
```
Comment (agiudiceandrea) [inline python/plugins/db_manager/db_model.py:403]: Not sure if this is the better way to check if the item is a raster layer in a GeoPackage container...

Comment (elpaso) [reply python/plugins/db_manager/db_model.py:403]: DB manager is ancient code, a more modern approach would be to use the connections API:

```python
md = QgsProviderRegistry.instance().providerMetadata('postgres')
conn = md.createConnection("dbname='tinyows_demo' host=localhost port=5433 username='***' password='***'", {})
table = conn.table('public', 'demo3dlinestring')
bool(table.flags() & QgsAbstractDatabaseProviderConnection.TableFlag.Raster)
bool(table.flags() & QgsAbstractDatabaseProviderConnection.TableFlag.Vector)
```

It should work with all QGIS supported DB providers (GPKG included).

**Comment by elpaso** [comment]
@agiudiceandrea see https://github.com/OSGeo/gdal/pull/10213

Maybe you can add a branch to check for the next GDAL version.

**Comment by elpaso** [comment]
> QgsGeoPackageProviderConnection

I will update `QgsGeoPackageProviderConnection` but a check for GDAL > 3.9.1 will be necessary.

**Comment by agiudiceandrea** [comment]
@elpaso, thanks again. I see that https://github.com/OSGeo/gdal/pull/10213 has not been backported to the GDAL 3.9 branch so, the check should be for GDAL >= 3.10.

Anyway I think it would be better if I remove the last commit [[db manager] Don't add a Rename action for raster layer in GeoPackage](https://github.com/qgis/QGIS/pull/57754/commits/164ba0283ad8f8ad64881cab937508f3f184b5e2) from this PR so this PR can be merged now.
I will then open a new PR to fix the renaming action for raster layer in GeoPackage after `QgsGeoPackageProviderConnection` will be updated.

**Comment by agiudiceandrea** [comment]
Hi @elpaso, I've removed the second commit from this PR. So it now only fixes the Delete functionality and can be safely merged and backported.
I'll open a new PR to also fix the Rename functionality with GDAL >= 3.10 thanks to your PRs https://github.com/OSGeo/gdal/pull/10213 and https://github.com/qgis/QGIS/pull/57812. The new PR cannot be backported since https://github.com/qgis/QGIS/pull/57812 has not been backported.

**Comment by agiudiceandrea** [comment]
To be backported also to release-3_38 branch, but the "backport release-3_38" tag is not available.

**Review by elpaso — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
> I'll open a new PR to also fix the Rename functionality with GDAL >= 3.10

PR: https://github.com/qgis/QGIS/pull/57865.

**Pull request merged**